hc/front/tests/test_add_channel.py

```diff
@@ -37,5 +37,24 @@ def test_instructions_work(self):
             r = self.client.get(url)
             self.assertContains(r, "Integration Settings", status_code=200)
 
+    def test_bad_kinds_not_working(self):
+        self.client.login(username="alice@example.org", password="password")
+        bad_kind = "twitter"
+        url = "/integrations/add_%s/" % bad_kind
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 404)
+         
     ### Test that the team access works
+    def test_can_add_channel_using_team_access(self):
+        url = "/integrations/add/"
+        form = {"kind": "email", "value": "bob@example.org"}
+
+        # Bob is in alice's team so he should be able to add a channel
+        self.client.login(username="bob@example.org", password="password")
+        r = self.client.post(url, form)
```

Comment:
Adding an integration does not sufficiently check that user registered with the email *bob@example.org* has  access to Alice's team. Use a different approach.